quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs

```diff
@@ -55,6 +59,14 @@ impl core::ops::Mul<Ratio<u64>> for Bandwidth {
     }
 }
 
+impl core::ops::Mul<Duration> for Bandwidth {
+    type Output = u64;
+
+    fn mul(self, rhs: Duration) -> Self::Output {
+        (self.bits_per_second * rhs.as_micros() as u64) / MICRO_BITS_PER_BYTE
```

Comment:
I went with an approach that tries to maintain resolution in the case we do not overflow, and otherwise divides first and then does a saturating multiplication. Let me know what you think